### PR TITLE
Add e2e integration tests for Postgres and remove 2dc runs

### PR DESCRIPTION
**Pull request by longquanzheng**
**What changed?**
Adding postgres to run with integration test suits

**Why?**
Follow up to a conversation in https://github.com/uber/cadence/pull/3488#discussion_r483220280 

**How did you test it?**
It's adding tests 

**Potential risks**
No

Local run succeeded:
```
docker-compose -f docker/buildkite/docker-compose-local.yml run integration-test-postgres
...
...
"service": "cadence-matching", "logging-call-at": "service.go:115"}
--- PASS: TestSizeLimitIntegrationSuite (14.23s)
    --- PASS: TestSizeLimitIntegrationSuite/TestTerminateWorkflowCausedBySizeLimit (2.44s)
PASS
coverage: 29.0% of statements in github.com/uber/cadence/common/..., github.com/uber/cadence/service/..., github.com/uber/cadence/client/..., github.com/uber/cadence/tools/...
ok  	github.com/uber/cadence/host	546.843s	coverage: 29.0% of statements in github.com/uber/cadence/common/..., github.com/uber/cadence/service/..., github.com/uber/cadence/client/..., github.com/uber/cadence/tools/...
1539.65user 518.08system 10:40.22elapsed 321%CPU (0avgtext+0avgdata 1528492maxresident)
136720inputs+1913456outputs (50major+10853577minor)pagefaults 0swaps
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -87,13 +87,13 @@ PKG_TEST_DIRS := $(filter-out $(INTEG_TEST_ROOT)%,$(TEST_DIRS))
 # Code coverage output files
 COVER_ROOT                 := $(BUILD)/coverage
 UNIT_COVER_FILE            := $(COVER_ROOT)/unit_cover.out
-INTEG_COVER_FILE           := $(COVER_ROOT)/integ_$(PERSISTENCE_TYPE)_cover.out
-INTEG_XDC_COVER_FILE       := $(COVER_ROOT)/integ_xdc_$(PERSISTENCE_TYPE)_cover.out
+INTEG_COVER_FILE           := $(COVER_ROOT)/integ_$(PERSISTENCE_TYPE)_$(PERSISTENCE_PLUGIN)_cover.out
```
Comment (longquanzheng) [inline Makefile:90]: I am still figuring out how to fix Makefile for buildkite. Suggestion is welcome.

**Comment by longquanzheng** [comment]
@yux0 since you are removing 2DC code, should I just remove all the 2dc stuff here? so that it will save you some effort to do it in another PR.

File: scripts/buildkite/gocov.sh [hunk 2]
```diff
@@ -8,10 +8,6 @@ go get github.com/dmetzgar/goveralls
 # download cover files from all the tests
 mkdir -p build/coverage
 buildkite-agent artifact download "build/coverage/unit_cover.out" . --step ":golang: unit test" --build "$BUILDKITE_BUILD_ID"
-buildkite-agent artifact download "build/coverage/integ_cassandra_cover.out" . --step ":golang: integration test with cassandra" --build "$BUILDKITE_BUILD_ID"
```
Comment (yux0) [inline scripts/buildkite/gocov.sh:11]: are those useless? I assume we don't generate code coverage in integration tests

**Comment by yux0** [comment]
> @yux0 since you are removing 2DC code, should I just remove all the 2dc stuff here? so that it will save you some effort to do it in another PR.

Thanks. It would be great

File: scripts/buildkite/gocov.sh (hunk 2 above)
Comment (longquanzheng) [reply scripts/buildkite/gocov.sh:11]: Yeah I believe so. 
And if you look at the list, it doesn't take ndc into account. I don't have permission, would you have help me kick off a run to see how the coverage would change?

**Review by andrewjdawson2016 — APPROVED** [review]
LGTM thanks @longquanzheng

**Comment by longquanzheng** [comment]
@yux0 @andrewjdawson2016 because I don't have permission :( can you help trigger a buildkite when reviewing it?

**Review by yux0 — APPROVED** [review]

**Pull request merged**